Cli key transfer (#1454)
ava-labs/avalanche-docs: docs/build/subnet/maintain/how-to-transfer-funds.md

---
tags: [Build, Subnets, Avalanche-CLI]
description:  If you need to send funds to your Subnet control key or need to move funds from one Ledger address index to another, this guide will demostrate how to enable direct transfers between ledger P-Chain addresses using the Avalanche-CLI command `avalanche key transfer`.
sidebar_label: Transfer P-Chain Funds
pagination_label: How to Use Avalanche-CLI to Transfer P-Chain Funds
sidebar_position: 3
---

# How to Use Avalanche-CLI to Transfer P-Chain Funds

> It is sometimes necessary to transfer funds between P-Chain wallets. This could be because:
> 
> 1. You need to send funds to your Subnet control key. It can end having no balance due to
> fee payments, and you need it to be funded to support Subnet operations.
> 2. You need to move funds from one Ledger address index to another. Sometimes a transfer
> to a ledger can be made to an address other than the one used by default by the CLI.

Reviewer: ```suggestion
Transferring funds between P-Chain wallets becomes necessary in certain situations:

1. Funds need to be sent to the Subnet control key, which might have a zero balance 
due to fee payments. The Subnet control key requires funding to ensure proper 
support for Subnet operations.
2. Funds need to be moved from one Ledger address index to another. 
Occasionally, a transfer made to a ledger can be made to an address different
from the default one used by the CLI.
```

To enable direct transfers between P-Chain addresses, use the command
`avalanche key transfer`. This operation involves a series of import/export
actions with the P-Chain and X-Chain. The fee for this operation is four times the typical
import operation fee, which comes out to 0.004 AVAX. You can find more

> information about fees [here](https://docs.avax.network/quickstart/transaction-fees).

Reviewer: ```suggestion
information about fees [here](/reference/standards/guides/txn-fees).
```

:::note

The `key transfer` command can also be applied to the stored keys managed by the CLI. It enables
moving funds from one stored key to another, and from a ledger to a stored key or the other way.

:::
This how-to guide focuses on transferring funds between ledger accounts.

## Prerequisites

> - [`Avalanche-CLI`](https://github.com/ava-labs/avalanche-cli) installed

Reviewer: ```suggestion
- [`Avalanche-CLI`](/tooling/cli-guides/install-avalanche-cli) installed
```
it is best practice to keep users in our docs instead of sending them to external links
